prow/crier/reporters/gcs/reporter.go

```diff
@@ -74,24 +75,73 @@ func (gr *gcsReporter) reportJobState(ctx context.Context, log *logrus.Entry, pj
 // happen before the pod itself gets to upload one, at which point the pod will
 // upload its own. If for some reason one already exists, it will not be overwritten.
 func (gr *gcsReporter) reportStartedJob(ctx context.Context, log *logrus.Entry, pj *prowv1.ProwJob) error {
-	s := downwardapi.PjToStarted(pj, nil)
+	bucketName, dir, err := util.GetJobDestination(gr.cfg, pj)
+	if err != nil {
+		return fmt.Errorf("failed to get job destination: %w", err)
+	}
+
+	if gr.dryRun {
+		log.WithFields(logrus.Fields{"bucketName": bucketName, "dir": dir}).Debug("Would upload started.json")
+		return nil
+	}
+
+	// Best-effort read of existing started.json; it's overwritten only if it's uploaded
+	// by crier and there is something new (clone record).
+	var existingStarted metadata.Started
+	content, err := io.ReadContent(ctx, log, gr.opener, providers.GCSStoragePath(bucketName, path.Join(dir, prowv1.StartedStatusFile)))
+	if err == nil {
```

Comment:
It feels a bit odd to check this condition first, when we do `if err != nil` pervasively in Go (like we're doing in line 118 below).

Reply:
done